sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -851,12 +851,21 @@ object DDLUtils {
     table.provider.isDefined && table.provider.get.toLowerCase(Locale.ROOT) != HIVE_PROVIDER
   }
 
-  def readHiveTable(table: CatalogTable): HiveTableRelation = {
-    HiveTableRelation(
-      table,
-      // Hive table columns are always nullable.
-      table.dataSchema.asNullable.toAttributes,
-      table.partitionSchema.asNullable.toAttributes)
+  def readHiveTable(catalog: SessionCatalog, table: CatalogTable): HiveTableRelation = {
```

Comment:
I can see `readHiveTable` is used by writing path (`InsertIntoStatement`) too. If we just get cached plan, will it be dangerous if the cached plan is out-of-date and Spark writes with incorrect metadata?

Reply:
I tried to reuse the cache of Datasource tables for Hive tables
InsertIntoStatement for Datasource tables, also fetches from the same cache. The cache invalidation have been taken care. 
From my reading, I didnt find any cases. Let me know if you find any cases that needs special handling. I will also check the code from this perspective again.